assets/js/components/ClusterDetails/ChecksResultOverview.jsx

```diff
@@ -1,36 +1,60 @@
 import React from 'react';
 import { format } from 'date-fns';
 
+import Spinner from '@components/Spinner';
 import CheckResultCount from './CheckResultCount';
 
 function ChecksResultOverview({
-  passing = 0,
-  warning = 0,
-  critical = 0,
-  lastCheckExecution = new Date(),
+  data,
+  error = null,
+  loading = false,
   onCheckClick,
 }) {
+  if (loading || data?.status === 'running') {
+    return (
+      <div className="flex flex-col items-center mt-2 px-4">
+        <Spinner />
+      </div>
+    );
+  }
+
+  if (error) {
+    return (
+      <div className="flex flex-col items-center mt-2 px-4">
+        <div className="text-center text-xs">{error}</div>
+      </div>
+    );
+  }
+
+  if (!data) {
+    return (
+      <div className="flex flex-col items-center mt-2 px-4">
+        <div className="text-center text-xs">No check results available.</div>
+      </div>
+    );
+  }
+
```

Comment:
Just a nitpick: I would pass individual values as props, just to reduce the dot notation access inside the component and avoid assuming a specific shape for the `data` object.

Also, you can avoid `if (loading || data?.status === 'running')` by doing

```js
<CheckResultOverview
  loading={loading || lastExecution.status === 'running'}
/>
```

when using the component

Reply:
Tech-debt ticket added to our backlog.
The purpose of this PR is simply to replace the old legacy code by the new